### PR TITLE
PF-1040: Surface google json exception errors

### PR DESCRIPTION
**Pull request by rogerwangcs**
Fix the create compute resource flights to throw `BadRequestException` instead of a 500 `InternalServerException` if there are issues with parameters in the flight's gcp request.

`GoogleJsonResponseException` returns 400 if a parameter was incorrectly formatted and 404 if a parameter value was not found, such as a gce machine type or cluster autoscaling policy uri. For both, surface the gcp error message and return a 400.

This also fixes a case where creating vertex ai notebooks with a bad machine type or vm image family will hang for all 10 flight retries before finally failing due to the 404 being retried.

Before:
```
{
	"jobReport": {
		"id": "e73805d7-a09f-47de-a27b-450d97a71857",
		"description": "Create controlled resource CONTROLLED_GCP_AI_NOTEBOOK_INSTANCE; id d08b97d5-217c-4221-b951-30be477a9f6f; name new-instance3",
		"status": "FAILED",
		"statusCode": 500,
		"submitted": "2023-08-25T20:38:19.627501Z",
		"completed": "2023-08-25T20:40:33.281971Z",
		"resultURL": "http://localhost:8080/api/workspaces/v1/63712477-55b9-49f2-b8db-ad127f458c83/resources/controlled/gcp/ai-notebook-instances/e73805d7-a09f-47de-a27b-450d97a71857/create-result"
	},
	"errorReport": {
		"message": "404 Not Found\nPOST https://notebooks.googleapis.com/v1/projects/terra-wsm-t-perky-corn-7923/locations/us-central1-a/instances?instanceId=sdgsdasfgasf\n{\n  \"code\": 404,\n  \"errors\": [\n    {\n      \"domain\": \"global\",\n      \"message\": \"Machine type asfafsa was not found\",\n      \"reason\": \"notFound\"\n    }\n  ],\n  \"message\": \"Machine type asfafsa was not found\",\n  \"status\": \"NOT_FOUND\"\n}",
		"statusCode": 500,
		"causes": []
	}
}
```
After:
```
{
	"jobReport": {
		"id": "479075ac-61bb-49a0-8776-14216380e20b",
		"description": "Create controlled resource CONTROLLED_GCP_AI_NOTEBOOK_INSTANCE; id 31b2aaa2-cd5c-4bda-9627-47799be83ab7; name new-instance3",
		"status": "FAILED",
		"statusCode": 400,
		"submitted": "2023-08-25T20:31:52.920250Z",
		"completed": "2023-08-25T20:32:04.519466Z",
		"resultURL": "http://localhost:8080/api/workspaces/v1/63712477-55b9-49f2-b8db-ad127f458c83/resources/controlled/gcp/ai-notebook-instances/479075ac-61bb-49a0-8776-14216380e20b/create-result"
	},
	"errorReport": {
		"message": "Machine type asfafsa was not found",
		"statusCode": 400,
		"causes": []
	}
}
```

### CODE REVIEW

**Comment by rogerwangcs** [comment]
@cbookg fyi

File: service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/gcp/gceinstance/CreateGceInstanceStep.java [hunk 1]
```diff
@@ -130,9 +131,10 @@ public StepResult doStep(FlightContext flightContext)
               zone,
               resource.getInstanceId());
           return StepResult.getStepResultSuccess();
-        } else if (HttpStatus.BAD_REQUEST.value() == e.getStatusCode()) {
-          // Don't retry bad requests, which won't change. Instead fail faster.
-          return new StepResult(StepStatus.STEP_RESULT_FAILURE_FATAL, e);
+        } else if (HttpStatus.BAD_REQUEST.value() == e.getStatusCode()
```
Comment (dexamundsen) [inline service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/gcp/gceinstance/CreateGceInstanceStep.java:134]: minor: consider a util function (may be under GcpResourceValidationUtils or GcpResourceConstants) to parse / handle GoogleJsonResponseException

**Review by dexamundsen — APPROVED** [review]

File: service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/gcp/gceinstance/CreateGceInstanceStep.java (hunk 1 above)
Comment (ddietterich) [reply service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/gcp/gceinstance/CreateGceInstanceStep.java:134]: +1

**Review by ddietterich — APPROVED** [review]
LGTM
I agree with Dex that the common filter would be worth extracting into a util method.

File: service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/gcp/gceinstance/CreateGceInstanceStep.java (hunk 1 above)
Comment (rogerwangcs) [reply service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/gcp/gceinstance/CreateGceInstanceStep.java]: Done

**Pull request merged**